optimade/models/utils.py

```diff
@@ -1,8 +1,100 @@
+import inspect
+import warnings
 import re
+from enum import Enum
+from typing import Optional
+
+from pydantic import Field
+
+_PYDANTIC_FIELD_KWARGS = inspect.signature(Field).parameters.keys()
 
 __all__ = ("CHEMICAL_SYMBOLS", "EXTRA_SYMBOLS", "ATOMIC_NUMBERS", "SemanticVersion")
 
 
+class SupportLevel(Enum):
+    MUST = "must"
+    SHOULD = "should"
+    OPTIONAL = "optional"
+
+
+def StrictField(
+    *args,
+    description: str = None,
+    **kwargs,
+) -> Field:
+    """A wrapper around `pydantic.Field` that does the following:
+
+    - Forbids any "extra" keys that would be passed to `pydantic.Field`,
+        unless they start with the `x-optimade` extension.
+    - Emits a warning when no description is provided.
+
+    Returns:
+        the pydantic field
+
+    """
+    # ban extra keywords from OPTIMADE fields
+    _banned = []
+    for key in kwargs:
+        if key not in set(
+            list(_PYDANTIC_FIELD_KWARGS)
+            + ["unit", "pattern", "uniqueItems", "support", "queryable", "sortable"]
```

Comment:
I feel like there must be a better way of doing this, but customising pydantic `Field`s doesn't seem well documented. This "works" and should catch any potential typos we could make when creating fields. It's a bit of a leaky abstraction...